onnxruntime/contrib_ops/cpu/transformers/beam_search_impl_t5.h

```diff
@@ -366,21 +386,13 @@ Status BeamSearchT5<T>::Execute(const FeedsFetchesManager& encoder_feeds_fetches
   }
 
   gsl::span<const float> final_beam_scores = beam_state.beam_scores;
-  if (this->IsCuda()) {
```

Comment:
I think we still need copy final beam scores to CPU unless you change this:
https://github.com/microsoft/onnxruntime/blob/30f4e7a8a988739d98dc5815a7a9aadcccbd1c00/onnxruntime/contrib_ops/cuda/transformers/beam_search.cc#LL31C63-L31C63